subgraphs/network/schema.graphql

```diff
@@ -0,0 +1,357 @@
+# Subgraph Schema: Layer One
+# Version: 1.0.0
+# See https://github.com/messari/subgraphs/blob/master/docs/Schema.md for details
+
+enum SubgraphNetwork {
+  ARBITRUM_ONE
+  ARWEAVE_MAINNET
+  AURORA
+  AVALANCHE
+  BOBA
+  BSC # aka BNB Chain
+  CELO
+  CLOVER
+  COSMOS
+  CRONOS
+  MAINNET # Ethereum Mainnet
+  FANTOM
+  FUSE
+  HARMONY
+  JUNO
+  MOONBEAM
+  MOONRIVER
+  NEAR_MAINNET
+  OPTIMISM
+  OSMOSIS
+  MATIC # aka Polygon
+  XDAI # aka Gnosis Chain
+}
+
+##########################
+#### Network Metadata ####
+##########################
+
+type Network @entity {
+  " Name of the network being indexed "
+  id: ID!
+
+  " The verison of the schema "
+  schemaVersion: String!
+
+  " The version of the subgraph "
+  subgraphVersion: String!
+
+  " The version of the methodology "
+  methodologyVersion: String!
+
+  #### Quantitative Data ####
+
+  " Number of cumulative unique authors (see type Author for explanation) "
+  cumulativeUniqueAuthors: Int!
+
+  " The total number of blocks in the blockchain "
+  blockHeight: Int!
+
+  " Total difficulty to reach the current block "
+  cumulativeDifficulty: BigInt!
+
+  " Total gas used "
+  cumulativeGasUsed: BigInt!
+
+  " Current gas limit "
+  gasLimit: BigInt!
+
+  " Total burnt fees on the network in native asset "
+  cumulativeBurntFees: BigInt!
+
+  " Total rewards emitted on the network in native asset "
+  cumulativeRewards: BigInt
+
+  " Total number of transactions that have occured on this network "
+  cumulativeTransactions: Int
+
+  " Current supply of the native asset "
+  totalSupply: BigInt
+
+  " Current blocks per day emission "
+  blocksPerDay: BigDecimal!
+
+  #### Snapshots ####
+
+  " Daily metrics for this blockchain "
+  dailySnapshots: [DailySnapshot!]! @derivedFrom(field: "network")
+
+  " Hourly metrics for this blockchain "
+  hourlySnapshots: [HourlySnapshot!]! @derivedFrom(field: "network")
+}
+
+###############################
+#### Blockchain Timeseries ####
+###############################
+
+type DailySnapshot @entity {
+  " { number of days since UNIX epoch time } "
+  id: ID!
+
+  " ID of the network this snapshot belongs to "
+  network: Network!
+
+  " Number of cumulative unique authors (see type Author for explanation) "
+  cumulativeUniqueAuthors: Int!
+
+  " Number of unique daily active authors "
+  dailyActiveAuthors: Int!
+
+  " Block height at this day (aka, blockNumber) "
+  blockHeight: Int!
+
+  " Timestamp of this snapshot "
+  timestamp: BigInt!
+
+  " Blocks created during this day (note, this is different from blocksPerDay) "
+  dailyBlocks: Int!
+
+  " Total difficulty up until this day "
+  cumulativeDifficulty: BigInt!
+
+  " Difficulty during this day "
+  dailyDifficulty: BigInt!
+
+  " Average difficulty for this day "
+  dailyMeanDifficulty: BigDecimal!
+
+  " Sum of all gas used during this day "
+  dailyCumulativeGasUsed: BigInt!
```

Comment:
is it possible to calculate cumulative fees as well?

Reply:
not possible on EVM b/c we don't have gasPrice: https://ethereum.org/ig/developers/docs/gas/#pre-london

from my understanding the burntFees in near are the transaction fees: https://near.org/blog/near-protocol-economics/

cosmos also needs gasPrice which we don't have: https://docs.cosmos.network/master/basics/gas-fees.html#introduction-to-gas-and-fees

Arweave: TODO

Reply:
ah, no problem